poms/bnd/pom.xml

```
@@ -0,0 +1,673 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.openhab.addons</groupId>
+  <artifactId>org.openhab.addons.reactor</artifactId>
+  <version>2.5.0-SNAPSHOT</version>
+
+  <packaging>pom</packaging>
+
+  <name>openHAB Add-ons</name>
+  <description>This project contains the add-ons of openHAB</description>
+
+  <organization>
+    <name>openHAB.org</name>
+    <url>http://www.openhab.org</url>
+  </organization>
+
+  <licenses>
+    <license>
+      <name>Eclipse Public License 2.0</name>
+      <url>https://www.eclipse.org/legal/epl-2.0/</url>
+    </license>
+  </licenses>
+
+  <scm>
+    <connection>scm:git:https://github.com/openhab/openhab2-addons.git</connection>
+    <developerConnection>scm:git:https://github.com/openhab/openhab2-addons.git</developerConnection>
+    <tag>HEAD</tag>
+    <url>https://github.com/openhab/openhab2-addons</url>
+  </scm>
+
+  <issueManagement>
+    <system>Github</system>
+    <url>https://github.com/openhab/openhab2-addons/issues</url>
+  </issueManagement>
+
+  <distributionManagement>
+    <repository>
+      <id>bintray</id>
+      <url>https://api.bintray.com/maven/openhab/mvn/openhab2-addons/;publish=1</url>
+    </repository>
+    <snapshotRepository>
+      <id>jfrog</id>
+      <url>https://openhab.jfrog.io/openhab/libs-snapshot-local</url>
+    </snapshotRepository>
+  </distributionManagement>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <oh.java.version>1.8</oh.java.version>
+    <maven.compiler.source>${oh.java.version}</maven.compiler.source>
+    <maven.compiler.target>${oh.java.version}</maven.compiler.target>
+    <maven.compiler.compilerVersion>${oh.java.version}</maven.compiler.compilerVersion>
+
+    <bnd.version>4.1.0</bnd.version>
+    <karaf.version>4.2.1</karaf.version>
```

We're currently using Karaf 4.2.2 in openHAB.

```suggestion
    <karaf.version>4.2.2</karaf.version>
```

---

How is the tooling's version related to your Karaf distribution version?
It should not matter as long as we don't introduce stuff that the distribution cannot work with.
That's for build time only -- or did you see another usage?

---

I don't know what the impact is of changing it.
The version seems mostly used in `features/karaf/pom.xml`.
There's also [bugfixes](https://issues.apache.org/jira/browse/KARAF-6021?jql=project%20%3D%20KARAF%20AND%20fixVersion%20%3D%204.2.2%20AND%20text%20~%20%22karaf-maven-plugin%22) in 4.2.2 for the karaf-maven-plugin ofcourse :-)

---

I'd also prefer 4.2.2 - using the same version reduces potential confusion and having more recent patch releases should generally be a best practice anyhow.

---

Hm, I follow the Karaf changes as good as I can all the time and I was not aware of any related bugfix.
But if you want a bump, you can get it.

---

Thanks for the bump! I still want to give compiling the code with Java 11 a try, so I might run into KARAF-6021.

---

> I still want to give compiling the code with Java 11 a try

That information has been the one I missed. :wink: 
If you had told me that 4.2.2 will improve the situation for Java 11 and you would like to use it, I just bumped it.
My reading has been "there are bugfixes between 4.2.1 and 4.2.2" but not that we need a special one resolved (there will always be some bugfixes).

---

No problem! I also want to give Karaf 4.2.3 a try. But I think spending time on this PR and the whole Bnd migration will have more added value. :slightly_smiling_face: